### ISSUE TITLE
Got Cura to build on Gentoo, but it crashes on startup

### ISSUE THREAD

**Issue by salfter** [Ultimaker/Cura#5010]
I'm trying to get Cura to build on Gentoo so that I don't have to keep using the appimage, but I first ran into issues with v3.2.1.  I'm trying again now with v3.6.0...found a few new dependencies and took care of those, but it's still segfaulting on startup.

**Application Version**
3.6.0

**Platform**
Gentoo Linux

**Qt**
5.11.1

**PyQt**
5.10.1

**Display Driver**
x11-drivers/xf86-video-amdgpu-18.1.0

**Steps to Reproduce**
```
sudo layman -a salfter # this is my overlay, with Cura and dependencies
sudo emerge -av cura
/usr/bin/cura # load my build, not the appimage in /usr/local/bin
```

**Actual Results**
```
[MainThread] UM.Resources.__initializeStoragePaths [414]: Initializing storage paths
[MainThread] UM.Resources.__initializeStoragePaths [425]: Config storage path is /home/salfter/.config/cura/master
[MainThread] UM.Resources.__initializeStoragePaths [433]: Data storage path is /home/salfter/.local/share/cura/master
[MainThread] UM.Resources.__initializeStoragePaths [445]: Cache storage path is /home/salfter/.cache/cura/master
[MainThread] UM.PackageManager.__init__ [49]: Found bundled packages JSON file: /usr/share/uranium/resources/bundled_packages/uranium.json
[MainThread] UM.PackageManager.__init__ [49]: Found bundled packages JSON file: /usr/share/cura/resources/bundled_packages/cura.json
[MainThread] UM.View.GL.OpenGLContext.detectBestOpenGLVersion [103]: Trying OpenGL context 4.1...
[MainThread] UM.View.GL.OpenGLContext.detectBestOpenGLVersion [119]: Trying OpenGL context 2.0...
[MainThread] UM.View.GL.OpenGLContext.detectBestOpenGLVersion [128]: We got at least OpenGL context 2.0: 3.1 No profile
[MainThread] UM.Qt.QtApplication.initialize [138]: Detected most suitable OpenGL context version: 2.0 No profile
[MainThread] UM.Qt.QtApplication.initialize [144]: Initializing job queue ...
[MainThread] UM.Qt.QtApplication.initialize [148]: Initializing version upgrade manager ...
[MainThread] UM.Settings.ContainerRegistry.addContainer [371]: Container [empty_definition_changes] added.
[MainThread] UM.Settings.ContainerRegistry.addContainer [371]: Container [empty_variant] added.
[MainThread] UM.Settings.ContainerRegistry.addContainer [371]: Container [empty_material] added.
[MainThread] UM.Settings.ContainerRegistry.addContainer [371]: Container [empty_quality] added.
[MainThread] UM.Settings.ContainerRegistry.addContainer [371]: Container [empty_quality_changes] added.
[MainThread] UM.PackageManager._loadManagementData [83]: Loaded bundled packages data from /usr/share/uranium/resources/bundled_packages/uranium.json
[MainThread] UM.PackageManager._loadManagementData [83]: Loaded bundled packages data from /usr/share/cura/resources/bundled_packages/cura.json
[MainThread] UM.PackageManager._loadManagementData [105]: Loaded user packages management file from /home/salfter/.local/share/cura/master/packages.json
[MainThread] UM.PackageManager._saveManagementData [117]: Package management file /home/salfter/.local/share/cura/master/packages.json was saved
[MainThread] UM.PluginRegistry.initializeBeforePluginsAreLoaded [92]: Loading plugin configuration file '/home/salfter/.config/cura/master/plugins.json'
2018-12-17 10:41:50,726 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin ConsoleLogger
2018-12-17 10:41:50,742 - INFO - [MainThread] CuraEngineBackend.CuraEngineBackend.__init__ [80]: Found CuraEngine at: /usr/bin/CuraEngine
2018-12-17 10:41:50,742 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin CuraEngineBackend
2018-12-17 10:41:50,747 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin UpdateChecker
2018-12-17 10:41:50,747 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin ConsoleLogger was already loaded
2018-12-17 10:41:50,750 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin LocalContainerProvider
2018-12-17 10:41:50,750 - INFO - [Thread-1] UpdateChecker.UpdateCheckerJob.run [36]: Checking for new version of cura
2018-12-17 10:41:50,753 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin SimpleView
2018-12-17 10:41:50,755 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin LocalFileOutputDevice
2018-12-17 10:41:50,756 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin FileLogger
2018-12-17 10:41:50,759 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin ScaleTool
2018-12-17 10:41:50,760 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin SelectionTool
2018-12-17 10:41:50,762 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin MirrorTool
2018-12-17 10:41:50,763 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin CameraTool
2018-12-17 10:41:50,766 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin RotateTool
2018-12-17 10:41:50,768 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin TranslateTool
2018-12-17 10:41:50,771 - WARNING - [MainThread] STLReader.STLReader.<module> [26]: Could not find numpy-stl, falling back to slower code.
2018-12-17 10:41:50,772 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin STLReader
2018-12-17 10:41:50,774 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin STLWriter
2018-12-17 10:41:50,775 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin OBJReader
2018-12-17 10:41:50,777 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin OBJWriter
2018-12-17 10:41:50,777 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin UpdateChecker was already loaded
2018-12-17 10:41:50,777 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin ConsoleLogger was already loaded
2018-12-17 10:41:50,777 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin LocalContainerProvider was already loaded
2018-12-17 10:41:50,778 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin SimpleView was already loaded
2018-12-17 10:41:50,778 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin LocalFileOutputDevice was already loaded
2018-12-17 10:41:50,778 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin FileLogger was already loaded
2018-12-17 10:41:50,778 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin ScaleTool was already loaded
2018-12-17 10:41:50,778 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin SelectionTool was already loaded
2018-12-17 10:41:50,778 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin MirrorTool was already loaded
2018-12-17 10:41:50,778 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin CameraTool was already loaded
2018-12-17 10:41:50,778 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin RotateTool was already loaded
2018-12-17 10:41:50,779 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin TranslateTool was already loaded
2018-12-17 10:41:50,779 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin STLReader was already loaded
2018-12-17 10:41:50,779 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin STLWriter was already loaded
2018-12-17 10:41:50,779 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin OBJReader was already loaded
2018-12-17 10:41:50,779 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin OBJWriter was already loaded
2018-12-17 10:41:50,781 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin GCodeWriter
2018-12-17 10:41:50,784 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin ImageReader
2018-12-17 10:41:50,787 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin UltimakerMachineActions
2018-12-17 10:41:50,789 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin XRayView
2018-12-17 10:41:50,792 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin FirmwareUpdater
2018-12-17 10:41:50,795 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin GCodeGzReader
2018-12-17 10:41:50,798 - WARNING - [MainThread] UM.Preferences.getValue [97]: Tried to get the value of non-existing setting cura/active_mode.
2018-12-17 10:41:50,798 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin PerObjectSettingsTool
2018-12-17 10:41:50,801 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin CuraProfileWriter
2018-12-17 10:41:50,803 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin MonitorStage
2018-12-17 10:41:50,812 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin USBPrinting
2018-12-17 10:41:50,824 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin UFPWriter
2018-12-17 10:41:50,828 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin RemovableDriveOutputDevice
2018-12-17 10:41:50,831 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin SupportEraser
2018-12-17 10:41:50,835 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin SolidView
2018-12-17 10:41:50,837 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin ChangeLogPlugin
2018-12-17 10:41:50,840 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin X3DReader
2018-12-17 10:41:50,843 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin SliceInfoPlugin
2018-12-17 10:41:50,846 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin PostProcessingPlugin
2018-12-17 10:41:50,848 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin UserAgreement
2018-12-17 10:41:50,851 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin FirmwareUpdateChecker
2018-12-17 10:41:50,853 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin VersionUpgrade30to31
2018-12-17 10:41:50,855 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin VersionUpgrade25to26
2018-12-17 10:41:50,858 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin VersionUpgrade22to24
2018-12-17 10:41:50,860 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin VersionUpgrade34to35
2018-12-17 10:41:50,862 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin VersionUpgrade33to34
2018-12-17 10:41:50,864 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin VersionUpgrade26to27
2018-12-17 10:41:50,867 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin VersionUpgrade21to22
2018-12-17 10:41:50,870 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin VersionUpgrade32to33
2018-12-17 10:41:50,872 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin VersionUpgrade27to30
2018-12-17 10:41:50,875 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin LegacyProfileReader
2018-12-17 10:41:50,881 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin GCodeReader
2018-12-17 10:41:50,885 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin 3MFWriter
2018-12-17 10:41:50,888 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin ModelChecker
2018-12-17 10:41:50,891 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin GCodeGzWriter
2018-12-17 10:41:50,893 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin CuraProfileReader
2018-12-17 10:41:50,893 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin CuraEngineBackend was already loaded
2018-12-17 10:41:50,896 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin GCodeProfileReader
2018-12-17 10:41:50,899 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin MachineSettingsAction
2018-12-17 10:41:50,903 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin XmlMaterialProfile
2018-12-17 10:41:50,908 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin SimulationView
2018-12-17 10:41:50,913 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin Toolbox
2018-12-17 10:41:50,940 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin UM3NetworkPrinting
2018-12-17 10:41:50,944 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin PrepareStage
2018-12-17 10:41:50,948 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [422]: Loaded plugin 3MFReader
2018-12-17 10:41:50,948 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin GCodeWriter was already loaded
2018-12-17 10:41:50,949 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin ImageReader was already loaded
2018-12-17 10:41:50,949 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin UltimakerMachineActions was already loaded
2018-12-17 10:41:50,949 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin XRayView was already loaded
2018-12-17 10:41:50,949 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin FirmwareUpdater was already loaded
2018-12-17 10:41:50,949 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin GCodeGzReader was already loaded
2018-12-17 10:41:50,949 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin PerObjectSettingsTool was already loaded
2018-12-17 10:41:50,950 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin CuraProfileWriter was already loaded
2018-12-17 10:41:50,950 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin MonitorStage was already loaded
2018-12-17 10:41:50,950 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin USBPrinting was already loaded
2018-12-17 10:41:50,950 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin UFPWriter was already loaded
2018-12-17 10:41:50,950 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin RemovableDriveOutputDevice was already loaded
2018-12-17 10:41:50,950 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin SupportEraser was already loaded
2018-12-17 10:41:50,950 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin SolidView was already loaded
2018-12-17 10:41:50,950 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin ChangeLogPlugin was already loaded
2018-12-17 10:41:50,950 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin X3DReader was already loaded
2018-12-17 10:41:50,951 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin SliceInfoPlugin was already loaded
2018-12-17 10:41:50,951 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin PostProcessingPlugin was already loaded
2018-12-17 10:41:50,951 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin UserAgreement was already loaded
2018-12-17 10:41:50,951 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin FirmwareUpdateChecker was already loaded
2018-12-17 10:41:50,951 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin VersionUpgrade30to31 was already loaded
2018-12-17 10:41:50,951 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin VersionUpgrade25to26 was already loaded
2018-12-17 10:41:50,951 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin VersionUpgrade22to24 was already loaded
2018-12-17 10:41:50,951 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin VersionUpgrade34to35 was already loaded
2018-12-17 10:41:50,952 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin VersionUpgrade33to34 was already loaded
2018-12-17 10:41:50,952 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin VersionUpgrade26to27 was already loaded
2018-12-17 10:41:50,952 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin VersionUpgrade21to22 was already loaded
2018-12-17 10:41:50,952 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin VersionUpgrade32to33 was already loaded
2018-12-17 10:41:50,952 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin VersionUpgrade27to30 was already loaded
2018-12-17 10:41:50,952 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin LegacyProfileReader was already loaded
2018-12-17 10:41:50,952 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin GCodeReader was already loaded
2018-12-17 10:41:50,952 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin 3MFWriter was already loaded
2018-12-17 10:41:50,952 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin ModelChecker was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin GCodeGzWriter was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin CuraProfileReader was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin CuraEngineBackend was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin GCodeProfileReader was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin MachineSettingsAction was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin XmlMaterialProfile was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin SimulationView was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin Toolbox was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin UM3NetworkPrinting was already loaded
2018-12-17 10:41:50,953 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin PrepareStage was already loaded
2018-12-17 10:41:50,954 - WARNING - [MainThread] UM.PluginRegistry.loadPlugin [378]: Plugin 3MFReader was already loaded
2018-12-17 10:41:50,957 - INFO - [MainThread] UM.VersionUpgradeManager.upgrade [130]: Looking for old configuration files to upgrade.
2018-12-17 10:41:50,958 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./packages.json, skip it
2018-12-17 10:41:50,959 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./plugins.json, skip it
2018-12-17 10:41:50,960 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./packages.json, skip it
2018-12-17 10:41:50,960 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./plugins.json, skip it
2018-12-17 10:41:50,961 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./packages.json, skip it
2018-12-17 10:41:50,961 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./plugins.json, skip it
2018-12-17 10:41:50,962 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./packages.json, skip it
2018-12-17 10:41:50,963 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./plugins.json, skip it
2018-12-17 10:41:50,964 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./packages.json, skip it
2018-12-17 10:41:50,964 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./plugins.json, skip it
2018-12-17 10:41:50,965 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./packages.json, skip it
2018-12-17 10:41:50,966 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./plugins.json, skip it
2018-12-17 10:41:50,966 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./packages.json, skip it
2018-12-17 10:41:50,967 - WARNING - [MainThread] UM.VersionUpgradeManager._getUpgradeTasks [281]: Failed to get file version: ./plugins.json, skip it
2018-12-17 10:41:50,970 - DEBUG - [MainThread] UM.Backend.Backend._logSocketState [178]: Socket state changed to Listening
2018-12-17 10:41:50,975 - INFO - [MainThread] UM.Backend.Backend.startEngine [74]: Started engine process: /usr/bin/CuraEngine
2018-12-17 10:41:50,975 - DEBUG - [MainThread] UM.Backend.Backend._backendLog [94]: [Backend] Calling engine with: ['/usr/bin/CuraEngine', 'connect', '127.0.0.1:49674', '']
2018-12-17 10:41:50,978 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] 
2018-12-17 10:41:50,978 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] Cura_SteamEngine version master
2018-12-17 10:41:50,979 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] Copyright (C) 2018 Ultimaker
2018-12-17 10:41:50,979 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] 
2018-12-17 10:41:50,979 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] This program is free software: you can redistribute it and/or modify
2018-12-17 10:41:50,979 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] it under the terms of the GNU Affero General Public License as published by
2018-12-17 10:41:50,980 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] the Free Software Foundation, either version 3 of the License, or
2018-12-17 10:41:50,980 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] (at your option) any later version.
2018-12-17 10:41:50,980 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] 
2018-12-17 10:41:50,981 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] This program is distributed in the hope that it will be useful,
2018-12-17 10:41:50,983 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] but WITHOUT ANY WARRANTY; without even the implied warranty of
2018-12-17 10:41:50,983 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
2018-12-17 10:41:50,983 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] GNU Affero General Public License for more details.
2018-12-17 10:41:50,984 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] 
2018-12-17 10:41:50,984 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] You should have received a copy of the GNU Affero General Public License
2018-12-17 10:41:50,984 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] along with this program.  If not, see <http://www.gnu.org/licenses/>.
2018-12-17 10:41:51,082 - DEBUG - [MainThread] UM.Backend.Backend._logSocketState [182]: Socket state changed to Connected
2018-12-17 10:41:51,082 - DEBUG - [MainThread] UM.Backend.Backend._onSocketStateChanged [172]: Backend connected on port 49674
2018-12-17 10:41:51,501 - DEBUG - [MainThread] CuraEngineBackend.CuraEngineBackend.slice [227]: Starting to slice...
2018-12-17 10:41:51,502 - WARNING - [MainThread] CuraEngineBackend.CuraEngineBackend.slice [231]: Slice unnecessary, nothing has changed that needs reslicing.
2018-12-17 10:41:51,965 - DEBUG - [MainThread] cura.Scene.CuraSceneController.setActiveBuildPlate [103]: Select build plate: 0
2018-12-17 10:41:51,966 - INFO - [MainThread] cura.CuraApplication.run [689]: Initializing variant manager
2018-12-17 10:41:51,969 - DEBUG - [MainThread] cura.Machines.VariantManager.initialize [59]: Exclude variant [empty_variant]
2018-12-17 10:41:51,970 - DEBUG - [MainThread] LocalContainerProvider.LocalContainerProvider.loadContainer [54]: Loading container ultimaker_s5_aluminum
2018-12-17 10:41:51,972 - DEBUG - [MainThread] UM.Settings.ContainerRegistry.addContainer [371]: Container [ultimaker_s5_aluminum] added.
2018-12-17 10:41:51,972 - DEBUG - [MainThread] LocalContainerProvider.LocalContainerProvider.loadContainer [54]: Loading container ultimaker_s5
2018-12-17 10:41:52,000 - DEBUG - [MainThread] UM.Settings.ContainerRegistry.addContainer [371]: Container [ultimaker_s5] added.
2018-12-17 10:41:52,001 - DEBUG - [MainThread] LocalContainerProvider.LocalContainerProvider.loadContainer [54]: Loading container ultimaker_s5_glass
2018-12-17 10:41:52,003 - DEBUG - [MainThread] UM.Settings.ContainerRegistry.addContainer [371]: Container [ultimaker_s5_glass] added.
2018-12-17 10:41:52,004 - INFO - [MainThread] cura.CuraApplication.run [693]: Initializing material manager
2018-12-17 10:41:52,093 - INFO - [MainThread] cura.CuraApplication.run [698]: Initializing quality manager
2018-12-17 10:41:52,101 - DEBUG - [MainThread] cura.Machines.QualityManager.initialize [136]: Lookup tables updated.
2018-12-17 10:41:52,101 - INFO - [MainThread] cura.CuraApplication.run [703]: Initializing machine manager
2018-12-17 10:41:52,102 - INFO - [MainThread] cura.CuraApplication.run [706]: Initializing container manager
2018-12-17 10:41:52,102 - INFO - [MainThread] cura.CuraApplication.run [709]: Initializing machine error checker
2018-12-17 10:41:52,165 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin._createView [279]: Creating post processing plugin view.
2018-12-17 10:41:52,166 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: BQ_PauseAtHeight
2018-12-17 10:41:52,167 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: ChangeAtZ
2018-12-17 10:41:52,167 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: DisplayFilenameAndLayerOnLCD
2018-12-17 10:41:52,168 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: ExampleScript
2018-12-17 10:41:52,168 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: FilamentChange
2018-12-17 10:41:52,169 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: PauseAtHeight
2018-12-17 10:41:52,169 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: PauseAtHeightRepRapFirmwareDuet
2018-12-17 10:41:52,169 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: PauseAtHeightforRepetier
2018-12-17 10:41:52,170 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: SearchAndReplace
2018-12-17 10:41:52,170 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin.loadScripts [169]: Begin loading of script: Stretch
2018-12-17 10:41:52,200 - DEBUG - [MainThread] UM.Qt.Bindings.Theme.load [172]: Loading theme file: /usr/share/cura/resources/themes/cura-light/theme.json
2018-12-17 10:41:52,206 - DEBUG - [MainThread] UM.Qt.Bindings.Theme.load [244]: Loaded theme /usr/share/cura/resources/themes/cura-light
2018-12-17 10:41:52,216 - DEBUG - [MainThread] PostProcessingPlugin.PostProcessingPlugin._createView [289]: Post processing view created.
2018-12-17 10:41:52,270 - DEBUG - [MainThread] LocalContainerProvider.LocalContainerProvider.loadContainer [54]: Loading container fdmprinter
2018-12-17 10:41:52,296 - DEBUG - [MainThread] UM.Settings.ContainerRegistry.addContainer [371]: Container [fdmprinter] added.
KCrash: Application 'python3.6m' crashing...
KCrash: Attempting to start /usr/lib64/libexec/drkonqi from kdeinit
sock_file=/var/run/user/1000/kdeinit5__1
2018-12-17 10:41:53,257 - DEBUG - [Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] [ERROR] Connection reset by peer
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib64/python3.6/logging/__init__.py", line 996, in emit
    self.flush()
  File "/usr/lib64/python3.6/logging/__init__.py", line 976, in flush
    self.stream.flush()
OSError: [Errno 9] Bad file descriptor
Call stack:
  File "/usr/lib64/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib64/python3.6/site-packages/UM/Backend/Backend.py", line 163, in _storeStderrToLogThread
    self._backendLog(line)
  File "/usr/lib64/python3.6/site-packages/UM/Backend/Backend.py", line 94, in _backendLog
    Logger.log("d", "[Backend] " + line_str.strip())
  File "/usr/lib64/python3.6/site-packages/UM/Logger.py", line 68, in log
    logger.log(log_type, message)
  File "/usr/lib/uranium/plugins/FileLogger/FileLogger.py", line 44, in log
    self._logger.debug(message)
Message: '[Thread-10] UM.Backend.Backend._backendLog [94]: [Backend] [ERROR] Connection reset by peer'
Arguments: ()
Exception in thread Thread-10:
Traceback (most recent call last):
  File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib64/python3.6/site-packages/UM/Backend/Backend.py", line 160, in _storeStderrToLogThread
    line = handle.readline()
OSError: [Errno 9] Bad file descriptor
```

**Comment by MathyV** [comment]
Hi, I was bitten by the same bug, this seems to be related to the version of PyQt5. Recently 5.9.2 was removed from the tree but Cura doesn't play nice yet with 5.10+

I was able to fix it by manually installing 5.9.2. You can find the ebuild here:
https://github.com/gentoo/gentoo/blob/f035df035f8640ed7ea4685105f0e7dd21aef51d/dev-python/PyQt5/PyQt5-5.9.2.ebuild

Also, there's a bug open in bugzilla talking about this:
https://bugs.gentoo.org/673220

PS: Cura is actually in the portage tree, and quite actively maintained, so your effort to create ebuilds would be more than welcome over there. Currently I have a pull request running for 3.6.0:

https://github.com/gentoo/gentoo/pull/10595

**Comment by salfter** [comment]
Copied that older PyQt5 into my overlay...it's building now.  There aren't a huge number of packages that depend on it, and none depend on v5.10, so I'm hoping that'll be enough to get it running.

Last time I looked for Cura in Portage, it was way behind.  I had my own working Cura ebuilds up to around v3.2.1, at which point it quit working and I switched over to using the appimage just to get it running again.  v3.6.0 has been out a while now, but the latest Cura ebuild in Portage is for v3.4.1.  I'm using v3.6.0 on Windows at the office, so it'd be nice to have the same version running everywhere.

FWIW, my ebuild (and its dependencies) are at https://gitlab.com/salfter/portage, and it's installable through Layman.

**Comment by salfter** [comment]
Looks like it's up and running now...got to the "add printer" dialog, which never happened before.  Sweet!

**Comment by zwim** [comment]
Hello Salfter,
I have added your overlay, cura-3.6.0 is building well, but if I try to emerge cura-4.0.0 portage is complaining about a missing file:


>>> Verifying ebuild manifests

!!! A file listed in the Manifest could not be found: /var/lib/layman/salfter/media-gfx/curaengine/files/curaengine-4.0.0-build-fix.patch


Could you please add the patch to your overlay.

Thanks
